pediatric_rootlets/analysis_comparison_spinal_vs_vertebral_levels.py

```diff
@@ -0,0 +1,278 @@
+    plt.legend(handles=handles, loc='lower left', ncol=2, fontsize=10, handlelength=1)
+
+    plt.savefig(f'{output_path}/boxplot_overlap_rootlets_vertebrae.svg', dpi=300)
+    plt.show()
+
+
+def create_scatterplot(df_overlap, output_path):
+    sns.set_palette('muted')
+    g = sns.FacetGrid(df_overlap, col='spinal_level_name', hue='sex', col_wrap=3, height=4, ylim=(0,100))
+    g.map(sns.scatterplot, 'age', 'overlap')
+    g.add_legend()
+    plt.subplots_adjust(top=0.9)
+    g.fig.suptitle('Scatterplots of Overlap by Age, Sex, and Spinal Level')
+    plt.savefig(f'{output_path}/scatterplot_overlap_rootlets_vertebrae.svg', dpi=300)
+    plt.show()
+
+
+def create_kdeplot(df_overlap, output_path):
+    sns.set_palette('muted')
+    g = sns.FacetGrid(df_overlap, col='spinal_level_name', hue='sex', col_wrap=3, height=4, ylim=(0,100))
+    g.map(sns.kdeplot, 'age', 'overlap', fill=True)
+    plt.subplots_adjust(top=0.9)
+    g.fig.suptitle('KDE plots of Overlap by Age, Sex, and Spinal Level')
+    plt.savefig(f'{output_path}/kdeplot_overlap_rootlets_vertebrae.svg', dpi=300)
+    plt.show()
+
+
+def main():
+    parser = get_parser()
+    args = parser.parse_args()
+
+    dir_path = os.path.abspath(args.i)
+    output_path = os.path.abspath(args.o)
+
+    if not os.path.isdir(dir_path):
+        print(f'ERROR: {args.i} does not exist.')
+        sys.exit(1)
+
+    df_participants = pd.read_csv(args.participants, sep='\t')
+    participants_age = df_participants[['participant_id', 'age']]
+    participants_sex = df_participants[['participant_id', 'sex']]
+
+    # Get all the CSV files in the directory generated by the 02a_rootlets_to_spinal_levels.py script
+    csv_files = glob.glob(os.path.join(dir_path, '**', '*pmj_distance_*[vertebral_disc|rootlets].csv'), recursive=True)
+
+    # if csv_files is empty, exit
+    if len(csv_files) == 0:
+        print(f'ERROR: No CSV files found in {dir_path}')
+
+    # Initialize an empty list to store the parsed data
+    parsed_data = []
+
+    # Loop across CSV files and aggregate the results into pandas dataframe
+    for csv_file in csv_files:
+        df_file = pd.read_csv(csv_file)
+        parsed_data.append(df_file)
+
+    # Combine list of dataframes into one dataframe
+    df = pd.concat(parsed_data)
+    df['age'] = df['fname'].apply(lambda x: participants_age[participants_age['participant_id'] == x[:7]]['age'].values[0])
+    df['age_group'] = df['age'].apply(
+        lambda x: '7-10' if x in AGE_GROUPS['7-10'] else '11-14' if x in AGE_GROUPS['11-14'] else '15-17')
+
+    # Extract rootlets or vertebrae level type from the fname and add it as a column
+    df['spinal_level_type'] = df['fname'].apply(
+        lambda x: 'rootlets' if 'label-rootlets' in x else 'vertebrae'
+    )
+
+    # Extract subjectID from the fname and add it as a column
+    df['subject'] = df['fname'].apply(lambda x: x.split('_')[0])
+
+    # Extract spinal level (cervical 2-8) and vertebral level (2-7)
```

Comment:
what do you think about this?

```suggestion
    # Keep spinal levels C2-C8) and vertebral level C2-C7
```